### PR TITLE
Patch Anomaly DLC

### PR DESCRIPTION
**Pull request by N7Huntsman**
## Additions
Dummy PR for tracking/testing purposes.
- Patch new Anomaly content to work with CE.

## Changes
- TBD

## References
Related to #3024 

## Reasoning
- Because the Cube wills it.

## Alternatives
None, really.

## Testing

Check tests you have performed:
- [ ] Compiles without warnings
- [ ] Game runs without errors
- [ ] (For compatibility patches) ...with and without patched mod loaded
- [ ] Playtested a colony (specify how long)

### CODE REVIEW

**Comment by SaltyKarl** [comment]
mortar shell Recipe's </fixedIngredientFilter> does not have Bioferrite in it

**Comment by SamaelGray** [comment]
I propose adding a bioferrite flamer fuel to the incinerator's own ammoset or a new one specific to it that deals high burn damage, more than prometheum and chemfuel, but does not spawn puddles. I can make the weapon distinct and keeps the bioferrite fuel of vanilla included in the CE version. The same fuel type can be used for the Hellcat too.

**Comment by SamaelGray** [comment]
The lab coat description mentions pockets, a +10 bulk capacity would make sense.

File: Anomaly/Defs/Ammo/NerveSpikerBolt.xml [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+		<graphicData>
+			<texPath>Things/Projectile/NerveSpikerShot</texPath>
+			<graphicClass>Graphic_Single</graphicClass>
+		</graphicData>
+		<projectile Class="CombatExtended.ProjectilePropertiesCE">
+			<damageDef>Nerve</damageDef>
+			<speed>34</speed>			
+			<damageAmountBase>12</damageAmountBase>
+			<armorPenetrationBlunt>7.08</armorPenetrationBlunt>
+			<armorPenetrationSharp>5</armorPenetrationSharp>
```
Comment (SamaelGray) [inline Anomaly/Defs/Ammo/NerveSpikerBolt.xml:60]: The sheet states 2.5mm, is this a typo or is it intended?
Plasteel bolts have 3.5mm AP so ~3mm would make sense unless there's a reason behind the 5.

File: Anomaly/Defs/Ammo/NerveSpikerBolt.xml [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+			<preExplosionSpawnChance>0.25</preExplosionSpawnChance>
```
Comment (SamaelGray) [inline Anomaly/Defs/Ammo/NerveSpikerBolt.xml:61]: Would bump to higher than 0.5.

File: Anomaly/Defs/Ammo/NerveSpikerBolt.xml [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+	<ThingDef ParentName="BaseCrossbowBoltProjectile">
+		<defName>Projectile_NerveSpikerBolt</defName>
+		<label>nerve spiker bolt</label>
+		<graphicData>
+			<texPath>Things/Projectile/NerveSpikerShot</texPath>
+			<graphicClass>Graphic_Single</graphicClass>
+		</graphicData>
+		<projectile Class="CombatExtended.ProjectilePropertiesCE">
+			<damageDef>Nerve</damageDef>
+			<speed>34</speed>
```
Comment (SamaelGray) [inline Anomaly/Defs/Ammo/NerveSpikerBolt.xml:57]: My revised formula states 28, the same formula currently in use for all arrows and bolts.

File: Anomaly/Patches/Misc/Mutants.xml [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+          <armorPenetrationBlunt>1.2</armorPenetrationBlunt>
+          <armorPenetrationSharp>0.6</armorPenetrationSharp>
+          <alwaysTreatAsWeapon>true</alwaysTreatAsWeapon>
+        </li>
+      </tools>
+    </value>
+  </Operation>
+
+  <!-- ========== Ghoul ========== -->
+  <Operation Class="PatchOperationReplace">
```
Comment (SamaelGray) [inline Anomaly/Patches/Misc/Mutants.xml:54]: The whole file is using the wrong indentation style.
I'd assume the ghouls have bioferrite claws and bump the sharp AP on their scratch attacks, probably increase the damage too. Something close to an unpowered power claw.
image

File: Anomaly/Patches/ThingDefs_Misc/Apparel_Packs.xml [hunk 5]
```diff
@@ -35,34 +34,122 @@
...
+		<value>
+			<thingClass>CombatExtended.ProjectileCE_SpawnsThing</thingClass>
+		</value>
+	</Operation>
+
+	<Operation Class="PatchOperationReplace">
+		<xpath>Defs/ThingDef[defName="Grenade_DisruptorFlare"]/projectile</xpath>
+		<value>
+		<projectile Class="CombatExtended.ProjectilePropertiesCE">
+			<dropsCasings>true</dropsCasings>
```
Comment (SamaelGray) [inline Anomaly/Patches/ThingDefs_Misc/Apparel_Packs.xml:50]: No need for casings as it's not a grenade, looks pretty much like a stick flare.

File: Anomaly/Patches/ThingDefs_Misc/Weapons_Ranged.xml [hunk 6]
```diff
@@ -0,0 +1,217 @@
...
+					<armorPenetrationBlunt>2.755</armorPenetrationBlunt>
+					<linkedBodyPartsGroup>Muzzle</linkedBodyPartsGroup>
+				</li>
+			</tools>
+		</value>
+	</Operation>
+
+	<!-- ========== Hellcat Rifle ========== -->
+
+	<Operation Class="CombatExtended.PatchOperationMakeGunCECompatible">
```
Comment (SamaelGray) [inline Anomaly/Patches/ThingDefs_Misc/Weapons_Ranged.xml:46]: To be switched to a Galil ACE in 5.56 low velocity ammoset with the appropriate texture.
image

File: Anomaly/Patches/ThingDefs_Misc/Weapons_Ranged.xml [hunk 7]
```diff
@@ -46,30 +46,30 @@
...
+			<SwayFactor>1.17</SwayFactor>
+			<Bulk>9.5</Bulk> <!-- Increased from base due to underbarrel weapon. -->
 		</statBases>
 		<Properties>
-			<recoilAmount>1.15</recoilAmount>
+			<recoilAmount>1.16</recoilAmount>
 			<verbClass>CombatExtended.Verb_ShootCE</verbClass>
 			<hasStandardCommand>true</hasStandardCommand>
-			<defaultProjectile>Bullet_545x39mmSoviet_FMJ</defaultProjectile>
+			<defaultProjectile>Bullet_556x45mmNATO_FMJ</defaultProjectile>
```
Comment (SamaelGray) [inline Anomaly/Patches/ThingDefs_Misc/Weapons_Ranged.xml:60]: Set to normal ammoset and projectile, should be the short-barrel (_SB) version

**Review by MaxDorob — DISMISSED** [review]
Code looks good

**Comment by SamaelGray** [comment]
Crossbow bolts and crossbows themselves like the Spiker are commonly considered medieval, wouldn't it be more appropriate for the Spike bolts' (bioferrite _arrows(?)_ bolts) texture path to be inside the Medieval folder?

File: Anomaly/Patches/ThingDefs_Races/Races_Entities_Misc.xml [hunk 8]
```diff
@@ -0,0 +1,669 @@
...
+	<Operation Class="PatchOperationAdd">
+		<xpath>Defs/ThingDef[defName="Noctol"]/statBases</xpath>
+		<value>
+			<MeleeDodgeChance>1</MeleeDodgeChance>
+			<MeleeCritChance>1</MeleeCritChance>
+			<MeleeParryChance>1</MeleeParryChance>
+		</value>
+	</Operation>
+
+	<Operation Class="PatchOperationAdd">
```
Comment (Tostov) [inline Anomaly/Patches/ThingDefs_Races/Races_Entities_Misc.xml:279]: Requires a replace op as it does start with armor already in vanilla so trying to patch add leaves them with vanilla armor value.

**Review by Tostov — CHANGES_REQUESTED** [review]
Overall seems workable, haven't run into any major bug issues from a decent amount of colony time (haven't tested all events in regular play but most have been completed in devmode quickly testing).
Entity downing seems to be a little odd since a lot of them tend to go down very quickly but just barely, and often get back up shortly afterwards. But otherwise not problematic as far as I have experienced.
As noted in the other note, noctol just needs its add changed to a replace and then I can't see any other specific issues with operation of the branch/patch.

**Review by Tostov — APPROVED** [review]

**Review by Tostov — CHANGES_REQUESTED** [review]
Deadlife pack currently does not consume charges.
In vanilla it is single use and disposable.
I'm not sure if we have a one use variant of the static launch verb we use on apparel.
If not we could just force it to use/reload with deadlife shells, which is sensible but does significantly increase the power of the pack (and consequentely reduce the effective strength of the creep joiner that has the deadlife ability).

**Review by Tostov — APPROVED** [review]

**Comment by Tostov** [comment]
All tested equipment/abilites seem to be working without issues now. Able to complete a playthrough without any gamebreaking issues arising outside of those now resolved.

**Pull request merged**